x/ccv/provider/keeper/key_assignment.go

```diff
@@ -567,3 +571,45 @@ func (k Keeper) DeleteKeyAssignments(ctx sdk.Context, chainID string) {
 		k.DeleteConsumerAddrsToPrune(ctx, chainID, consumerAddrsToPrune.VscId)
 	}
 }
+
+// KeyAssignmentValidateBasic validates all the genesis state for key assignment
+func KeyAssignmentValidateBasic(
+	assignedKeys []types.ValidatorConsumerPubKey,
+	byConsumerAddrs []types.ValidatorByConsumerAddr,
+	consumerAddrsToPrune []types.ConsumerAddrsToPrune,
+) error {
+	for _, e := range assignedKeys {
+		if strings.TrimSpace(e.ChainId) == "" {
+			return sdkerrors.Wrap(ccvtypes.ErrInvalidGenesis, "consumer chain id must not be blank")
+		}
+		if err := sdk.VerifyAddressFormat(e.ProviderAddr); err != nil {
+			return sdkerrors.Wrap(ccvtypes.ErrInvalidGenesis, fmt.Sprintf("invalid provider address: %s", e.ProviderAddr))
+		}
+		if e.ConsumerKey == nil {
```

Comment:
Is there any validation you can do with the key value besides just confirming the pointer is non-nil? Maybe a helper exists in sdk or tm

Reply:
Hmm I'm not sure to be honest. It's basic validation only, a full semantic validation that all the data is consistent logically isn't the goal here so we can probably leave it like this.